app/src/main/java/com/example/habithelper/activities/HabitDetailsActivity.java

```diff
@@ -0,0 +1,147 @@
+                Intent i = new Intent(HabitDetailsActivity.this, MainActivity.class);
+                i.putExtra("tab", "habits");
+                startActivity(i);
+                finish();
+            }
+        });
+    }
+
+    /**
+     * Determines whether the user has been tracking habits long enough to analyze their past data
+     * If so, calls a method to determine and display their most impactful habits
+     * If not, displays a card to tell the user to continue tracking.
+     *
+     */
+    private void getMostImpactfulHabits() {
+        ParseQuery<TrackDay> query = ParseQuery.getQuery(TrackDay.class);
+        query.include(TrackDay.KEY_PARENT_USER);
+        query.include(TrackDay.KEY_DATE_NUMBER);
+        query.include(TrackDay.KEY_TRACK_ARRAY);
+        query.include(TrackDay.KEY_MOOD);
+        query.whereEqualTo(TrackDay.KEY_PARENT_USER, currentUser);
+        query.addDescendingOrder(TrackDay.KEY_DATE_NUMBER);
+        query.findInBackground(new FindCallback<TrackDay>() {
+            @Override
+            public void done(List<TrackDay> daysTracked, ParseException e) {
+                if (e != null) {
+                    return;
+                }
+                numDaysTracked = daysTracked.size();
+
+                // ensures the number of days the user has tracked is greater than the number of habits they are tracking
+                // required for LeastSquaresRegression
+                if (numDaysTracked > numHabits) {
+                    averageMood = getAverageMood(daysTracked);
+                    //TODO: set visibility of percentage calculator!
+                    lrc = new LinearRegressionCalculator();
+                    // calculates the array of correlation coefficients of each habit on the user's mood
+                    lrc.performLeastSquares(daysTracked, numDaysTracked, numHabits);
+                    double[] resultsArray = lrc.leastSquaresResult;
+                    int habitIndex = findIndexOfHabit();
+                    for (int i = 0; i< resultsArray.length; i++){
+                        System.out.println(resultsArray[i]);
+                    }
+                    habitCoefficient = resultsArray[habitIndex];
+                    percentChange = ((int) Math.round(100 * ((double) habitCoefficient / averageMood)));
+                    System.out.println(percentChange);
+                } else {
+                    //TODO: what to do if the user hasn't tracked enough days yet
+                }
+            }
+        });
+    }
+
+    /**
+     * Finds the index of the desired habit in the habit list
+     * @return the habit's index
+     */
+    private int findIndexOfHabit() {
+        int index = -1;
+        for (int i = 0; i<numHabits; i++){
+            if(habitsList.get(i).equals(habit.getHabitName())){
+                return i;
+            }
+        }
+        return index;
+    }
+
+    /**
+     * Determines and displays the user's average mood
```

Comment:
this comment similarly to line 123 is redundant since  it literally translates the function name

Reply:
it also is not correct since this method doesn't "display" mood